packages/mobile/src/send/utils.ts

```diff
@@ -93,13 +99,41 @@ function dailySpent(now: number, recentPayments: PaymentInfo[]) {
   return amount
 }
 
-export function isPaymentLimitReached(
-  now: number,
-  recentPayments: PaymentInfo[],
-  initial: number
-): boolean {
-  const amount = dailySpent(now, recentPayments) + initial
-  return amount > DAILY_PAYMENT_LIMIT_CUSD
+export function useDailyTransferLimitValidator(
+  amount: BigNumber,
+  currency: CURRENCY_ENUM
+): [boolean, () => void] {
+  const dispatch = useDispatch()
+
+  const exchangeRatePair = useSelector(exchangeRatePairSelector)
+
+  const dollarAmount = useMemo(() => {
+    if (currency === CURRENCY_ENUM.DOLLAR) {
+      return amount
+    } else {
+      const exchangeRate = getRateForMakerToken(
+        exchangeRatePair,
+        CURRENCY_ENUM.DOLLAR,
+        CURRENCY_ENUM.GOLD
+      )
+      return goldToDollarAmount(amount, exchangeRate) || new BigNumber(0)
+    }
+  }, [amount])
+
+  const recentPayments = useSelector(getRecentPayments)
+  const localCurrencyExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+
+  const now = Date.now()
```

Comment:
Nit nit: if you aren't reusing `now`, you can call `Date.now` in the function argument to avoid this extra variable/line

Reply:
It's actually being used twice (lines 129 & 133) :)